alpha/apps/kaltura/modules/extwidget/actions/thumbnailAction.class.php

```diff
@@ -453,8 +453,8 @@ public function execute()
 		}
 		
 		$nocache = false;
-		if ($secureEntryHelper->shouldDisableCache() || kApiCache::hasExtraFields() ||
-			(!$secureEntryHelper->isKsWidget() && $secureEntryHelper->hasRules(ContextType::THUMBNAIL)))
+		if (!$secureEntryHelper->hasKESRule() && ($secureEntryHelper->shouldDisableCache() || kApiCache::hasExtraFields() ||
```

Comment:
I would probably change all the places where you named the functions or variables from the word KES to remoteServer (i.e hasKESRule hasServeRemoteServerRule). KES is not a terminology the Kaltura server should know about.